packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -136,29 +192,44 @@ export const useGridSelection = (apiRef: GridApiRef, props: GridComponentProps):
 
       const hasCtrlKey = event.metaKey || event.ctrlKey;
 
-      // Without checkboxSelection, multiple selection is only allowed if CTRL is pressed
-      const isMultipleSelectionDisabled = !checkboxSelection && !hasCtrlKey;
-      const resetSelection = !canHaveMultipleSelection || isMultipleSelectionDisabled;
-
-      if (resetSelection) {
-        apiRef.current.selectRow(
-          params.id,
-          hasCtrlKey || checkboxSelection ? !apiRef.current.isRowSelected(params.id) : true,
-          true,
-        );
+      if (event.shiftKey && (canHaveMultipleSelection || checkboxSelection)) {
+        apiRef.current.expandRowRangeSelection(params.id);
```

Comment:
This method could be only available inside the hook.

Reply:
As explained in https://github.com/mui-org/material-ui-x/pull/2456#discussion_r718209180, I did not find a clean way to keep it inside the `useGridSelection` hook

I could remove it from the documentation.
But for me it's not that strange to have a method saying "apply the range selection logic between this row and the last interacted row".

Reply:
The problem is that its use case is too specific.